File: src/j1bwmfinddifference/Java/com/example/bwmfinddifference/DifferenceFinder.java
// File: DifferenceFinder.java

package com.example.bwmfinddifference;

public class DifferenceFinder {
    public static int extraCharacterIndex(String str1, String str2) {
        int result = 0;

        for (int i = 0; i < str1.length(); i++) {
            result ^= str1.charAt(i);
        }

        for (int i = 0; i < str2.length(); i++) {
            result ^= str2.charAt(i);
        }

        return str1.length() > str2.length() ? str1.indexOf((char)result) : str2.indexOf((char)result);
    }
}

File: src/j1bwmfinddifference/TS/differenceFinder.ts
// File: differenceFinder.ts

export function extraCharacterIndex(str1: string, str2: string): number {
    let result = 0;

    for (let i = 0; i < str1.length; i++) {
        result ^= str1.charCodeAt(i);
    }

    for (let i = 0; i < str2.length; i++) {
        result ^= str2.charCodeAt(i);
    }

    return str1.length > str2.length
        ? str1.indexOf(String.fromCharCode(result))
        : str2.indexOf(String.fromCharCode(result));
}
